libcudacxx/test/libcudacxx/cuda/annotated_ptr/utils.h

```diff
@@ -33,8 +35,10 @@ __device__ __host__ void assert_rt_wrap(cudaError_t code, const char* file, int
   if (code != cudaSuccess)
   {
 #ifndef TEST_COMPILER_NVRTC
-    printf("assert: %s %s %d\n", cudaGetErrorString(code), file, line);
-#endif
+    NV_IF_ELSE_TARGET(NV_IS_DEVICE,
+                      (printf("assert: %s %s %d\n", cudaGetErrorString(code), file, line);),
```

Comment:
The order is wrong and `cudaGetErrorString()` still ends up being called on the device.

Reply:
🤦‍♂️